### ISSUE TITLE
fields annotated with Sling Models annotations should be marked as implicitly written

### ISSUE THREAD

**Issue by santiagoGPNC** [DmytroTroynikov/aemtools#249]
When a variable in a sling model (either a field or a constructor parameter) is annotated with any of the annotations in _org.apache.sling.models.annotations.injectorspecific_ (e.g. @ValueMapValue), it would be great if this plugin marked those fields as implicitly written so we don't get a warning about them.

The same could be done for the @Reference annotation in OSGi components.

I think a new implicitUsageProvider can be provided to check for such cases.

**Comment by koha26** [comment]
Hi @santiagoGPNC, thank you very much for reporting your suggestion. I will prioritize it and fix it in next releases.

**Comment by koha26** [comment]
Hi @santiagoGPNC, I started working on your issue and I have a question regarding variables in a sling model (constructor parameter). Could you please provide an example of code with injecting via constructor?

Because I tested the following case and didn't get the mentioned warning.
```
@Model(adaptables = Resource.class)
public class SlingModelA {

    private final String text;
    private final String author;

    @Inject
    public SlingModelA(@ValueMapValue(name = JcrConstants.JCR_CREATED_BY) final String createdBy,
                                    @ValueMapValue(name = "text") final String text,
                                    @OSGiService final ServiceA serviceA) {
        this.text = text;
        this.author = serviceA.getAuthor(createdBy).orElse(null);
    }
}
```

The same is not reproduced for method injection.

**Comment by santiagoGPNC** [comment]
hi @koha26  this happens when you use field injection.
e.g.

    @Model(adaptables = Resource.class)
     public class ModelA  {

        @SlingObject
        private Resource resource;

        @ChildResource(name = "child")
        private List<Resource> childResource;

        @ValueMapValue(name = "configuration")
        private String configuration;

        @OSGiService
        private ModelFactory modelFactory;
    }


In this case IntelliJ will complain with a warning like "Private field 'configuration' is never assigned ".
You can click a button to make IntelliJ treat fields annotated as written, but it has to be done for every annotation.

From what I see every Sling models annotation is annotated itself with @InjectAnnotation. maybe that is something that can be used to identify them.

Thank you very much for your effort.

**Comment by koha26** [comment]
@santiagoGPNC , this issue has been released in the latest version of plugin v1.0.4.1.

**Comment by santiagoGPNC** [comment]
thank you @koha26 . I'm giving it a try and it seems to be working. thanks!
